frontend/src/app-components/dialogs/DeleteDialog.tsx

```diff
@@ -17,16 +17,36 @@ import {
 } from "@mui/material";
 
 import { DialogTitle } from "@/app-components/dialogs/DialogTitle";
+import { useDelete } from "@/hooks/crud/useDelete";
+import { useDeleteMany } from "@/hooks/crud/useDeleteMany";
 import { DialogControl } from "@/hooks/useDialog";
 import { useTranslate } from "@/hooks/useTranslate";
+import { EntityType } from "@/services/types";
+import { IEntityMapTypes } from "@/types/base.types";
 
 export type DeleteDialogProps<T = string> = DialogControl<T>;
 export const DeleteDialog = <T extends any = string>({
   open,
-  callback,
   closeDialog: closeFunction,
-}: DeleteDialogProps<T>) => {
+  data: ids,
+  callback,
+  entity = EntityType.ATTACHMENT,
+  onDeleteError = () => {},
+  onDeleteSuccess = () => {},
+}: DeleteDialogProps<T> & {
+  entity?: keyof IEntityMapTypes;
+  onDeleteError?: (error: Error) => void;
+  onDeleteSuccess?: (data?: unknown) => void;
+}) => {
   const { t } = useTranslate();
+  const { mutateAsync: deleteEntity } = useDelete(entity, {
+    onError: onDeleteError,
+    onSuccess: onDeleteSuccess,
+  });
+  const { mutateAsync: deleteEntities } = useDeleteMany(entity, {
+    onError: onDeleteError,
+    onSuccess: onDeleteSuccess,
```

Comment:
Don't we need to clear the selection onSuccess ? `setDate([])`

Reply:
Done :heavy_check_mark: